All Git & JS tasks - Done

KolomoetsAndrey opened the pull request:
Git tasks - Done

KolomoetsAndrey committed:
Add files via upload

Git tasks - Done

KolomoetsAndrey committed:
arraySum

реализовать функцию которая вернет сумму элементов произвольного массива

KolomoetsAndrey committed:
Update and rename one.js to jsTasks.js

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,18 @@
+    for (var i = 0; i < array.length; i++) {
+        sum += array[i];
+    }
+    console.log(sum);
```
из условия: "которая вернет сумму элементов", твоя же функция ничего не возвращает, а выводит в консоль сумму

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update index.html

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
Update jsTasks.js

KolomoetsAndrey committed:
update jsTasks.js

Найдено более логичное решение задания №3

KolomoetsAndrey committed:
Update jsTasks.js

Задание 4: перемещение переменной arrFibonacci внутрь функции;

KolomoetsAndrey committed:
Update jsTask.js

избавляемся от внешних переменных

KolomoetsAndrey committed:
Update jsTasks.js

Переименование значений для "понятности" и "чистоты кода".

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+// 1. Реализовать функцию которая вернет сумму элементов произвольного массива:
+
+function arraySum(array) {
```
имя метода не совсем корректное, ты получаешь сумму, соотв метод будет называться аля "получить что-то"

KolomoetsAndrey replied:
согласен, исправлю

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+
+function arraySum(array) {
+    var sum = 0;
+    for (var firsfElement = 0; firsfElement < array.length; firsfElement++) {
```
это не **firsfElement**, это индекс, счетчик, но не элемент

KolomoetsAndrey replied:
согласен, переименую (музыкой наверное навеяло :)

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+// или
+
+var sortAscendind = [4, 1, 45, 6, 37, 554, 2];
+sortAscendind.sort(function (firsfElement, secondElement) {
```
можно было не дублировать :)

KolomoetsAndrey replied:
гугл подсказал несколько вариантов решения, какой из них верен хотел бы понять...

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+// b) цикл for:
+
+function sortFunction() {
+    var arr = [4, 1, 45, 6, 37, 554, 2];
```
массив, хорошо бы, передать, никто не смодет использовать твою функцию для своего массива

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+
+// b) цикл for:
+
+function sortFunction() {
```
лучше бы **getSortedArray**

KolomoetsAndrey replied:
согласен, подправлю

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+
+function sortFunction() {
+    var arr = [4, 1, 45, 6, 37, 554, 2];
+    for (a = 0; a < arr.length; a++) {
```
о оператор **var/let/const** ? и имя a ничего не говорит

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+function sortFunction() {
+    var arr = [4, 1, 45, 6, 37, 554, 2];
+    for (a = 0; a < arr.length; a++) {
+        for (b = 0; b < arr.length; b++) {
```
этот цикл можно крутить меньшее количество раз и опять же, **var/let/const** + имя b ничего не говорит

KolomoetsAndrey replied:
с циклом как-то не подружился, буду разбираться, по ходу исправлю

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+
+function uniqueElements(arr) {
+    var result = [];
+    nextInput:
```
метки не пишем, плохая практика, тк ломает логический порядок кода :(

KolomoetsAndrey replied:
разобрался, справляю

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+function uniqueElements(arr) {
+    var result = [];
+    nextInput:
+    for (var firsfElement = 0; firsfElement < arr.length; firsfElement++) {
```
это не **firsfElement**, это индекс, счетчик...

KolomoetsAndrey replied:
согласен, исправляю

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+    nextInput:
+    for (var firsfElement = 0; firsfElement < arr.length; firsfElement++) {
+        var tempElement = arr[firsfElement];
+        for (var secondElement = 0; secondElement < result.length; secondElement++) {
```
**secondElement** это не элемент, а индекс

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+    for (var firsfElement = 0; firsfElement < arr.length; firsfElement++) {
+        var tempElement = arr[firsfElement];
+        for (var secondElement = 0; secondElement < result.length; secondElement++) {
+            if (result[secondElement] == tempElement) continue nextInput;
```
можно, как вариант, выставить флаг в **true**, если не нашел такой элемент и по этому условию добавить в результирующий массив новый элемент

vadimk7 commented on jsTasks.js:
```diff
@@ -0,0 +1,87 @@
+    for (var firsfElement = 0; firsfElement < quantity; firsfElement++) {
+        arrFibonacci[firsfElement + 2] = arrFibonacci[firsfElement] + arrFibonacci[firsfElement + 1];
+    }
+    return arrFibonacci;
```
ну, вернет quantity + 2 элементов

KolomoetsAndrey replied:
вот тут не совсем понял, если можно чуть подробней, если нет то буду искать выход и исправлять по ходу понимания

KolomoetsAndrey committed:
update jsTask.js

задание 1: переименованы названия индексов и функций согласно требования ментора;
задание 2: переименованы названия функций и убрано дублирование одного из вариантов реализации (метод sort);
задание 3: изменено решение задачи (избавился от меток);
задание 4: изменено решение задачи;
